### PR TITLE
Remove decimal drop down 

### LINKED ISSUE #148: Do not require users to choose a data type for Decimals
- [ ] Optional Decimal fields render as number, null dropdown
 - [ ] Decimal fields render as number
 - [ ] When a user saves the json file, the number fields are saved as strings. For example `"weight": "3.25"`
 - [ ] When a user loads a json file with a string input, it will fill in the field correctly

### PR DESCRIPTION
**Pull request by micahwoodard**
fixes #148 
- removes drop down widget if options are string, number

### CODE REVIEW

**Comment by micahwoodard** [comment]
Hi @helen-m-lin and @mekhlakapoor I'm definitely not done but so far I've created a custom select widget and basically if the options == ['string', 'number'], the selectwidget doesn't appear. It's working in that aspect but the formatting now seems off for the rest of the select widgets. I was wondering if you had any suggestions on why or how to change it back to match the rest of the widgets? I'm going to keep digging but figured I'd ask if you knew off the top of your head!

File: src/custom-ui/CustomWidgets.js [hunk 1]
```diff
@@ -3,6 +3,7 @@ import 'react-datetime/css/react-datetime.css'
 import moment from 'moment'
 import RadioWidget from '@rjsf/material-ui/lib/RadioWidget/RadioWidget'
 import CheckboxWidget from '@rjsf/material-ui/lib/CheckboxWidget/CheckboxWidget'
+import SelectWidget from '@rjsf/material-ui/lib/SelectWidget/SelectWidget'
```
Comment (helen-m-lin) [inline src/custom-ui/CustomWidgets.js:6]: The select widget format can be fixed by using the rjsf/core package! I believe using radio and checkbox from material-ui was a custom case; the default theme used is bootstrap3 from rjsf/core.

```suggestion
import SelectWidget from '@rjsf/core/lib/components/widgets/SelectWidget'
```

File: src/custom-ui/CustomWidgets.js [hunk 2]
```diff
@@ -76,4 +77,19 @@ CustomTextWidget.propTypes = {
...
+ * Return no SelectWidget when options
+ * are string and number
+ */
+const CustomSelectWidget = (props) => {
+  const options = []
+  for (let i = 0; i < props.options.enumOptions.length; i++) {
+    options.push(props.options.enumOptions[i].label)
+  }
+  if (options.sort().join() !== ['number', 'string'].join()) {
+    const newProps = { ...props, hideLabel: true }
```
Comment (helen-m-lin) [inline src/custom-ui/CustomWidgets.js:90]: Could you check that the number input field is the one that is always shown on default by rjsf? I'm concerned that if the string input is shown in some cases, users would not have an option to change it once the select dropdown is hidden.

**Review by helen-m-lin — COMMENTED** [review]
Left some comments! Could you also check that the output json passes pydantic validation?

**Comment by micahwoodard** [comment]
I added test scripts and  checked that all anyOfs have number first. I think I need to add another layer if the options are number, string, null though so currently doing that

**Comment by micahwoodard** [comment]
Okay added in condition of removing string if options are number/string/null

**Comment by micahwoodard** [comment]
> Left some comments! Could you also check that the output json passes pydantic validation?

Yes it seems like it validates, but I'm basing that on the fact that when I click the validate button, it doesn't complain. Is that what you mean?

**Comment by micahwoodard** [comment]
John mentioned that the text box actually needs to be a string text box so I'm working on that now

**Comment by micahwoodard** [comment]
@helen-m-lin I've spent a bit of time last week and this morning trying to figure out how to either change the resulting textbox or change the default of the anyof widget but I haven't had a ton of luck. Is there anything you would suggest looking into or would changing the schema to have the string be the first input be a possibility?

**Pull request closed without merging**